RFC-011-SWAP-Omni-Layer-basic.md

```diff
@@ -0,0 +1,329 @@
+
+### Dust
+
+Bitcoin Core has limitations on minimum amounts owned by a UTXO. The minimum value is 546 satoshis for a normal transaction and 294 satoshis for SegWit transactions.
+See [RFC-010](./RFC-010-Omni-Layer.md#dust) for details.
+
+The values above will be referred as `min_sat` from now on.
+
+`OP_RETURN` outputs are not spendable and hence SHOULD not be assigned any Bitcoin value. Only spendable outputs MUST have at least `min_sat` Satoshis assigned.
+
+### HTLC Address
+
+The HTLC address MUST either be:
+1. the P2WSH address of the HTLC, as described in [RFC-005](./RFC-005-SWAP-Bitcoin-basic.md)
+2. the P2SH of the P2WSH above, as descriped in [BIP-0141 Segregated Witness](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki)
+
+(1) is preferred.
+However, not all Omni Layer wallets implementation support bech32 addresses.
+Hence, (2) can be used as a fallback.
+
+The simple send MUST NOT be done to the direct P2SH address of the HTLC.
+
+This choice has been made to:
+- limit the number of addresses to watch, facilitating implementation
+- avoid defining the basic HTLC twice (once, in P2SH format, once in P2WSH format)
+- follow the common strategy for segwit backward compatibility (nesting in P2SH)
+
+### Inputs & Outputs format to spend the HTLC
+
+This section describes specific inputs & outputs to construct an Omni Layer transaction for the *redeeming* and *refunding* of an HTLC.
+This is to facilitate the constructions and filtering of such transactions for the purpose of a basic swap.
+
+To construct a transaction to spend from the HTLC, the following information is needed:
+
+| Variable           | Description                                                                                                 |
+|:---                |:---                                                                                                         |
+| `htlc_utxo`        | UTXO locked by the HTLC, created with the deployment transaction                                            |
+| `to_address`       | Address to which the Omni Layer token is sent out of the HTLC                                               |
+| `change_address`   | Address to which the Bitcoin change will be sent upon redeeming; MUST be different than `to_address`        |
+| `change_utxo`      | UTXO locked by `change_address` that owns some Bitcoin asset to pay for the mining fee                      |
+| `init_change_sat`  | Amount of Satoshis owned by `change_utxo` before the redeem                                                 |
+| `mining_sat`       | The amount of Satoshi to be spent on mining fees                                                            |
+| `omni_data`        | The Omni Layer data that transfers the Omni Layer Asset as described in [Omni Layer data](#omni-layer-data) |
+
+ The Bitcoin transaction MUST be constructed with the following inputs, outputs and values.
+ The order of the inputs and outputs MUST be as described.
```

Comment:
Are you saying that 
* Output 1: is always the `target_address`
* Output 2: is always the `change_address`
* Output 3: is always `OP_RETURN?`

Obviously, we could ensure this is done in our code but I'm not sure if we should make this a general requirement, also, imho btsieve should still be able to find transactions not following this standard. 

Additionally, you mention that `change_address` may be omitted, meaning, output 2 will be `OP_RETURN`.

Reply:
Two things:
1. This is how to construct, not how to watch. See in `execution` part for how to watch.
2. Yes all you are saying is correct.

Reply:
I'm not sure if I understand what you are getting at, and I think the order of the outputs is not fixed and should not matter. 

Look at this transaction for instance: https://www.blockchain.com/btc/tx/79858587cb04fe293efdec9ccb8fb2bb8b4a7bf7751df2c9bb7c084386a98973?show_adv=true 

- output1= `change_address`
- output2= `op_return`
- output3= `to_address`

Reply:
So you are just saying to remove this then? `The order of the inputs and outputs MUST be as described.`

Reply:
Yes, at least I don't understand why it is required.